src/AlgebraicGeometry/ToricVarieties/CohomologyClasses/constructors.jl

```diff
@@ -54,11 +54,17 @@ julia> cohomology_class(d)
 Cohomology class on a normal toric variety given by 6*x3
 ```
 """
-function cohomology_class(d::ToricDivisor)
+function cohomology_class(d::ToricDivisor; quick::Bool = false)
+  if quick == false
     indets = gens(cohomology_ring(toric_variety(d)))
     coeff_ring = coefficient_ring(toric_variety(d))
     poly = sum(coeff_ring(coefficients(d)[k]) * indets[k] for k in 1:length(indets))
     return CohomologyClass(toric_variety(d), poly)
+  end
+  indets = gens(parent(cohomology_ring(toric_variety(d), check = false)))
+  coeff_ring = coefficient_ring(toric_variety(d))
+  poly = cohomology_ring(toric_variety(d))(sum(coeff_ring(coefficients(d)[k]) * indets[k] for k in 1:length(indets)))
+  return CohomologyClass(toric_variety(d), poly, quick = quick)
```

Comment:
```suggestion
  return CohomologyClass(toric_variety(d), poly, true)
```
Let's be explicit here, since we at this point know that `quick` is `true`.